### PR TITLE
Move identity to sdk/identity

### PR DESCRIPTION
**Pull request by mitchdenny**
This PR lifts and shifts the ```identity/client``` library into ```sdk/identity/azure-identity``` path as part of our overall relocation work.

### CODE REVIEW

File: pom.client.xml [hunk 1]
```diff
@@ -703,6 +703,6 @@
     <module>./eventhubs/client</module>
     <module>./keyvault/client</module>
     <module>./sdk/tracing</module>
-    <module>./identity/client</module>
+    <module>./sdk/identity/azure-identity</module>
```
Comment (g2vinay) [inline pom.client.xml]: So, we are not using `client` for track 2 sdks @JonathanGiles

Comment (JonathanGiles) [reply pom.client.xml]: Unfortunately not. The pattern is `microsoft-azure-` is track one, `azure-` is track two.

Comment (weshaggard) [reply pom.client.xml]: Yes we are trying to use the package name and not client or data-plane monikers

**Pull request merged**